sushichef.py

```diff
@@ -45,35 +49,84 @@ class MyChef(SushiChef):
-            For example, add the command line option   lang="fr"  and the string
-            "fr" will be passed along to `construct_channel` as kwargs['lang'].
-        Returns: ChannelNode
-        """
-        channel = self.get_channel(*args, **kwargs)  # Create ChannelNode from data in self.channel_info
+        LOGGER.info("Constructing channel from {}...".format(BASE_URL))
+        channel = self.get_channel(*args, **kwargs)                         # Creates ChannelNode from data in self.channel_info
+        LOGGER.info('   Writing {} Folder...'.format(CHANNEL_NAME))    
+        endpoint = BASE_URL + "content/"
+        scrape_content(endpoint, channel)
+        raise_for_invalid_channel(channel)                                  # Check for errors in channel construction
+        return channel
 
-        # TODO: Replace next line with chef code
-        raise NotImplementedError("constuct_channel method not implemented yet...")
+""" Helper Methods """
+###########################################################
 
-        raise_for_invalid_channel(channel)  # Check for errors in channel construction
+def read_source(url):
+    html = downloader.read(url)    
+    return BeautifulSoup(html, "html.parser")    
 
-        return channel
+def replace_all(text, replacements):
+    for condition, replacement in replacements.items():
+        text = text.replace(condition, replacement)
+    return text
+
+def scrape_content(endpoint, channel, existingNode=None):
+    replacements = {" ": "%20", "#": "%23"} 
+    content = read_source(endpoint)
+    attributes = content.find("tbody").find_all("td", "text-xs-left")
+
+    for attribute in attributes:
+        source_id = attribute.attrs["data-sort-value"]
+
+        # Check if it is mp4 file
+        if source_id.endswith(".mp4"):
+            video_info = attribute.find("a")
+            video_title = str(video_info.string)
+            filter_video_link = video_info.attrs["href"][1:].replace(" ", "%20")
+            video_link = BASE_URL + filter_video_link
+            video_file = files.VideoFile(path=video_link)
+            video_node = nodes.VideoNode(
+                source_id=source_id,
+                title=video_title,
+                files=[video_file],
+                license=CHANNEL_LICENSE
+            )
+            existingNode.add_child(video_node)         
+
+        # Check if it is a directory
+        if source_id.startswith("dir"):
+            title = str(attribute.find("strong").string)
+            topic_node = nodes.TopicNode(source_id=source_id, title=title)
+            if existingNode:
+                existingNode.add_child(topic_node)
+            else:
+                channel.add_child(topic_node)                            
 
+            new_end_point = replace_all(title, replacements)
+            new_end = endpoint + "{}/".format(new_end_point)
+            scrape_content(new_end, channel, topic_node)
```

Comment:
Could you also have an `else` statement for any unrecognized formats and use `LOGGER.error` to log it? Might be useful for discovering any edge cases